platforms/android/tangram/src/main/java/com/mapzen/tangram/MapController.java

```diff
@@ -1134,4 +1138,37 @@ String getFontFallbackFilePath(int importance, int weightHint) {
         return fontFileParser.getFontFallback(importance, weightHint);
     }
 
+    // Tmp Asset Path
+    // ==============
+    String copyAssetToTmpPath(String path) {
+
+        String tmpDir = mapView.getContext().getExternalCacheDir().getAbsolutePath();
+
+        try {
+            InputStream in = assetManager.open(path);
+            File outFile = new File(tmpDir, path);
+
+            if (!outFile.exists()) {
```

Comment:
I agree its a tricky thing to solve but I think it will be a very rare case.

The path returned from `getExternalCacheDir` is: `/storage/emulated/0/Android/data/com.mapzen.tangram.android/cache`, so the asset's path comes out to be `/storage/emulated/0/Android/data/com.mapzen.tangram.android/cache/new-york_new-york.mbtiles` for example. Its the same path http cache is saved.